gnd/src/main/java/com/google/android/gnd/ui/projectselector/ProjectSelectorViewModel.java

```diff
@@ -15,48 +15,88 @@
  */
 package com.google.android.gnd.ui.projectselector;
 
+import android.util.Log;
+
 import androidx.lifecycle.LiveData;
 import androidx.lifecycle.MutableLiveData;
+
 import com.google.android.gnd.repository.DataRepository;
 import com.google.android.gnd.repository.Resource;
+import com.google.android.gnd.rx.Result;
 import com.google.android.gnd.system.AuthenticationManager;
 import com.google.android.gnd.ui.common.AbstractViewModel;
 import com.google.android.gnd.vo.Project;
-import io.reactivex.Completable;
+
+import io.reactivex.Observable;
+import io.reactivex.subjects.PublishSubject;
+
 import java.util.Collections;
 import java.util.List;
+
 import javax.inject.Inject;
 
 public class ProjectSelectorViewModel extends AbstractViewModel {
   private static final String TAG = ProjectSelectorViewModel.class.getSimpleName();
 
   private final DataRepository dataRepository;
   private final MutableLiveData<Resource<List<Project>>> projectSummaries;
-  private final AuthenticationManager authManager;
+  private final PublishSubject<Integer> projectSelections;
+  public final MutableLiveData<Project> activeProject;
+  public final MutableLiveData<Throwable> activateProjectErrors;
+  public final Observable<Result<Project>> activeProjectStream;
 
   @Inject
   ProjectSelectorViewModel(DataRepository dataRepository, AuthenticationManager authManager) {
     this.dataRepository = dataRepository;
     this.projectSummaries = new MutableLiveData<>();
-    this.authManager = authManager;
-  }
+    this.activeProject = new MutableLiveData<>();
+    this.activateProjectErrors = new MutableLiveData<>();
+    this.projectSelections = PublishSubject.create();
+
+    Observable<AuthenticationManager.User> user = authManager.getUser();
```

Comment:
Are we sure it's safe to do all this work in the constructor? Or should it be run in a separate method and invoked from the appropriate lifecycle?

Reply:
For this particular case, I think it makes sense--the list of available projects needs to be fetched as soon as the VM is instantiated, as these are passed as an original list of options to the Fragment. Thereafter, we wait for selections over that list using the PublishSubject defined here.

Instead of being a stream, I've changed this to refer to only the first user returned by the authManager, to prevent inadvertent actions as other users (not that this is likely).

This also enabled me to be rid of those calls to `Result`.

Open to a better approach if you have one!